File: 프로그래머스/unrated/181830. 정사각형으로 만들기/정사각형으로 만들기.java
//1. 기본 풀이.
class Solution {
    public int[][] solution(int[][] arr) {
        //1. N 찾기. 
        int N = arr.length;
        for(int i = 0; i < arr.length; i++){
            N = Math.max(arr[i].length,N);
        }
        int[][] answer = new int[N][N];
          //  나머지는 범위 내인 경우만 값 받기
        for(int i = 0; i < N; i++){
            if(i >= arr.length) break;
            for(int j = 0; j < N; j++){
                if(j >= arr[i].length) break;
                answer[i][j] = arr[i][j];
            }
        }
        return answer;
    }
}

File: 프로그래머스/unrated/181830. 정사각형으로 만들기/정사각형으로 만들기.cs

using System;

public class Solution {
    public int[,] solution(int[,] arr) 
    {
        int max = Math.Max(arr.GetLength(0), arr.GetLength(1));
        int[,] answer = new int[max, max];

        for(int y = 0; y < arr.GetLength(0); ++y)
        {
            for(int x = 0; x < arr.GetLength(1); ++x)
                answer[y, x] = arr[y, x];
        }

        return answer;
    }
}